### PR TITLE
Magic attach guide

### PR DESCRIPTION
**Pull request by lucasmoura**
## Proposed Commit Message
docs: add magic attach docs for api endpoints

Add docs explaining how to use the magic attach endpoints.
Additionally, add extra step on the magic attach test to see the wait output for a invalid token

## Test Steps
Verify if the docs are covering all it should for the magic attach endpoints
Verify if the new step introduced on the magic attach test is working as expected

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [ ] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: dev-docs/howtoguides/how_to_use-magic_attach_endpoints.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+       "environment_vars": []}
+    },
+    "attributes": {
+      "expires": "EXPIRE_DATE",
+      "expires_in": SECONDS_TO_EXPIRE, 
+      "token": "MAGIC_ATTACH_TOKEN",
+      "user_code": "USER_CODE"
+    }
+    "type": "MagicAttachInitiate"
+  },
```
Comment (CalvoM) [inline dev-docs/howtoguides/how_to_use-magic_attach_endpoints.md:37]: A small linting error. I think there is some json linting error.

Comment (lucasmoura) [reply dev-docs/howtoguides/how_to_use-magic_attach_endpoints.md]: Good catch. I have just fixed it

File: dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
+# How to use magic attach API endpoints
+
+Right now, the pro client provides three distinct endpoints to make it easier to perform
```
Comment (orndorffgrant) [inline dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md:3]: The docs are intended to always be up to date, so I don't think we need the "Right now"
Also `pro` isn't released yet, and these docs will be immediately available, so we should use `ua`.
```suggestion
The UA Client provides three distinct endpoints to make it easier to perform
```

File: dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
+# How to use magic attach API endpoints
+
```
Comment (orndorffgrant) [inline dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md:2]: Let's include a minimum version notice
```suggestion
> **Notice:**
> Minimum version: 27.11
```

File: dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+We will explain how to use each endpoint and what is the expected output for each of them.
+
+## Initiate endpoint
+
+To start the magic attach flow, we need to create a token for it. The initiate endpoint
+will perform exactly that. When you run:
+
+```console
+$ pro api u.pro.attach.magic.initiate.v1
```
Comment (orndorffgrant) [inline dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md:18]: I think we'll replace all of this with `pro` after it's released
```suggestion
$ ua api u.pro.attach.magic.initiate.v1
```

File: dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+is the information that will be presented to the user, which it will make possible for the user
+to validate the magic attach on the pro portal. Additionally, the `token` information is required
+for the other two API endpoints which will be described next.
+
+
+## Wait endpoint
+
+After we initiate the magic attach procedure, the user must go to the pro portal and validate
+the `user_code` it received. Once that is done, a pro token will be generated for the user, allowing
+the attach procedure to begin. The wait endpoint allow us to wait for the user to perform all those
```
Comment (orndorffgrant) [inline dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md:55]:
```suggestion
the attach procedure to begin. The wait endpoint will wait for the user to perform all of those
```

File: dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+steps on the pro portal. To call it, use:
+
+```console
+$ pro api u.pro.attach.magic.initiate.v1 --args magic_token=MAGIC_ATTACH_TOKEN
+```
+
+Note here that the command requires the `token` that was generated in the initiate step. After
+calling this command, we will keep pooling to see if there is any updates on that token. If the
```
Comment (orndorffgrant) [inline dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md:63]: let's make it clearer that the wait command does the polling for you
```suggestion
Note here that the command requires the `token` that was generated in the initiate step.
This command will block and poll the server until there are any updates for that token. If the
```

File: dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+It is expected that the token will be valid for about 10 minutes. Therefore, we expect the wait
+command to keep polling for about that amount of time.
+
+
+## Revoke
+
+If we want to revoke the token create during the initiate call, we can use the revoke command:
+
+```console
+$ ua pro u.pro.attach.magic.revoke.v1 --args magic_token=MAGIC_ATTACH_TOKEN
```
Comment (orndorffgrant) [inline dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md:122]:
```suggestion
$ ua api u.pro.attach.magic.revoke.v1 --args magic_token=MAGIC_ATTACH_TOKEN
```

**Review by orndorffgrant — COMMENTED** [review]
Just some nits - looks good!

**Review by lucasmoura — COMMENTED** [review]
@orndorffgrant do you think we should also rename `pro portal` to `advantage portal` here ?

**Comment by orndorffgrant** [comment]
@lucasmoura Yes I think so for now

**Comment by orndorffgrant** [comment]
@lucasmoura I think my comments from last review are still open (at least on the current version on github) :upside_down_face:

**Comment by lucasmoura** [comment]
:facepalm: I don't how I did not added those changes on the previous commit. Hopefully it is fixed now

File: dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+the `user_code` it received. Once that is done, a ua token will be generated for the user, allowing
+the attach procedure to begin. The wait endpoint will wait for the user to perform all of those
+steps on the advantage portal. To call it, use:
+
+```console
+$ ua api u.pro.attach.magic.initiate.v1 --args magic_token=MAGIC_ATTACH_TOKEN
+```
+
+Note here that the command requires the `token` that was generated in the initiate step. This command
+will block and pool the server until there are any updates for that token. If the
```
Comment (renanrodrigo) [inline dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md:66]: `s/pool/poll/`?

File: dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+}
+```
+
+It is expected that the token will be valid for about 10 minutes. Therefore, we expect the wait
+command to keep polling for about that amount of time.
+
+
+## Revoke
+
+If we want to revoke the token create during the initiate call, we can use the revoke command:
```
Comment (renanrodrigo) [inline dev-docs/howtoguides/how_to_use_magic_attach_endpoints.md:122]: `s/create/created/`?

**Review by renanrodrigo — APPROVED** [review]
Just a couple typos
LGTM

**Comment by lucasmoura** [comment]
@renanrodrigo updated

**Pull request merged**